securedrop_client/gui/widgets.py

```diff
@@ -3238,64 +3752,7 @@ def on_reply_sent(self, source_uuid: str, reply_uuid: str, reply_text: str) -> N
         self.reply_flag = True
         if source_uuid == self.source.uuid:
             self.add_reply_from_reply_box(reply_uuid, reply_text)
-
-
-class SourceConversationWrapper(QWidget):
-    """
-    Wrapper for a source's conversation including the chat window, profile tab, and other
-    per-source resources.
-    """
-
-    def __init__(self, source: Source, controller: Controller) -> None:
-        super().__init__()
-
-        self.source_uuid = source.uuid
-        controller.source_deleted.connect(self._on_source_deleted)
-        controller.source_deletion_failed.connect(self._on_source_deletion_failed)
-
-        # Set layout
-        layout = QVBoxLayout()
-        self.setLayout(layout)
-
-        # Set margins and spacing
-        layout.setContentsMargins(0, 0, 0, 0)
-        layout.setSpacing(0)
-
-        # Create widgets
-        self.conversation_title_bar = SourceProfileShortWidget(source, controller)
-        self.conversation_view = ConversationView(source, controller)
-        self.reply_box = ReplyBoxWidget(source, controller)
-        self.waiting_delete_confirmation = QLabel("Deleting...")
-        self.waiting_delete_confirmation.setObjectName("SourceConversationWrapper_source_deleted")
-        self.waiting_delete_confirmation.hide()
-
-        # Add widgets
-        layout.addWidget(self.conversation_title_bar)
-        layout.addWidget(self.conversation_view)
-        layout.addWidget(self.reply_box)
-        layout.addWidget(self.waiting_delete_confirmation, alignment=Qt.AlignCenter)
-
-        # Connect reply_box to conversation_view
-        self.reply_box.reply_sent.connect(self.conversation_view.on_reply_sent)
-        self.conversation_view.conversation_updated.connect(
-            self.conversation_title_bar.update_timestamp
-        )
-
-    @pyqtSlot(str)
-    def _on_source_deleted(self, source_uuid: str):
-        if self.source_uuid == source_uuid:
-            self.conversation_title_bar.hide()
-            self.conversation_view.hide()
-            self.reply_box.hide()
-            self.waiting_delete_confirmation.show()
-
-    @pyqtSlot(str)
-    def _on_source_deletion_failed(self, source_uuid: str):
-        if self.source_uuid == source_uuid:
-            self.waiting_delete_confirmation.hide()
-            self.conversation_title_bar.show()
-            self.conversation_view.show()
-            self.reply_box.show()
+            self.update_deletion_markers(self.source.collection)
```

Comment:
i haven't tested this but i think we have to be careful about passing the source collection directly here without a try-catch, see https://github.com/freedomofpress/securedrop-client/issues/906

Reply:
I've added the exception handling we talked about.